src/test/java/project/bookstore/service/BookServiceTest.java

```diff
@@ -0,0 +1,221 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import project.bookstore.dto.book.BookDto;
+import project.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import project.bookstore.dto.book.BookSearchParameters;
+import project.bookstore.dto.book.CreateBookRequestDto;
+import project.bookstore.mapper.BookMapper;
+import project.bookstore.model.Book;
+import project.bookstore.model.Category;
+import project.bookstore.repository.book.BookRepository;
+import project.bookstore.repository.book.BookSpecificationBuilder;
+import project.bookstore.service.impl.BookServiceImpl;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @Mock
+    private BookSpecificationBuilder bookSpecificationBuilder;
+
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    private Book book;
+    private CreateBookRequestDto requestDto;
+    private BookDto bookDto;
+
+    @BeforeEach
+    void setUp() {
+        book = new Book();
+        book.setId(1L);
+        book.setTitle("Book Title");
+        book.setAuthor("Author");
+        book.setPrice(BigDecimal.valueOf(10.95));
+        book.setIsbn("9781244567897");
+        Category category = new Category();
+        book.setCategories(Set.of(category));
+
+        requestDto = new CreateBookRequestDto();
+        requestDto.setTitle("Book Title");
+        requestDto.setAuthor("Author");
+        requestDto.setPrice(BigDecimal.valueOf(10.95));
+        requestDto.setIsbn("9781244567897");
+
+        bookDto = new BookDto();
+        bookDto.setTitle("Book Title");
+        bookDto.setAuthor("Author");
+        bookDto.setPrice(BigDecimal.valueOf(10.95));
+        bookDto.setIsbn("9781244567897");
+        bookDto.setCategories(Set.of(category));
+    }
+
+    @Test
+    @DisplayName("""
+            Create a new book and save it to database
+            """)
+    void save_ValidRequestDto_ReturnsBook() {
+        when(bookMapper.toModel(requestDto)).thenReturn(book);
+        when(bookRepository.save(book)).thenReturn(book);
+
+        Book actual = bookService.save(requestDto);
+        Book expected = book;
+        assertEquals(expected, actual);
+        verify(bookRepository, times(1)).save(book);
+    }
+
+    @Test
+    @DisplayName("""
+            Create a new book with invalid request dto and save it to database
+            """)
+    void save_InvalidRequestDto_ReturnsException() {
+        doThrow(new IllegalArgumentException("Invalid request DTO"))
+                .when(bookMapper).toModel(requestDto);
```

Comment:
do you have such logic in your mapper?  I can't find it. So, what are you testing here?